buildscripts/kokoro/xds.sh

```diff
@@ -20,15 +20,24 @@ popd
 git clone -b "${branch}" --single-branch --depth=1 https://github.com/grpc/grpc.git
 
 grpc/tools/run_tests/helper_scripts/prep_xds.sh
+
+# Test cases "path_matching" and "header_matching" are not included in "all",
+# because not all interop clients in all languages support these new tests.
+#
+# TODO: remove "path_matching" and "header_matching" from --test_case after
+# they are added into "all".
 JAVA_OPTS=-Djava.util.logging.config.file=grpc-java/buildscripts/xds_logging.properties \
   python3 grpc/tools/run_tests/run_xds_tests.py \
-    --test_case=all \
+    --test_case="all,path_matching,header_matching" \
     --project_id=grpc-testing \
-    --source_image=projects/grpc-testing/global/images/xds-test-server \
+    --source_image=projects/grpc-testing/global/images/xds-test-server-2 \
```

Comment:
What's the difference between the two images?

Reply:
`-2`'s java server has changes from #7207, for the new RPCs.
Images cannot be updated (can only delete and recreate). So had to create this new one.
Plan is to update the old image when all tests use the new one. And switch it back when removing `path_matching,header_matching` from `--test_case`.